### PR TITLE
[in progress] Update dependencies, add possibility to work with tcp

### PR DESCRIPTION
**Pull request by s-mage**
This PR is just hotfixes for getting things done (I wanted to push messages through tcp and I've added the code), but in general it all could be done more neatly. For example, for now there is hardcoded port and only one atom for all the system. Plus it's a logger, right? All log tasks should be ran asynchronously not to stop main code execution.

### CODE REVIEW

**Comment by s-mage** [comment]
https://github.com/t0xa/gelfj/blob/master/src/main/java/org/graylog2/GelfMessage.java#L106 ah, I did it wrong... tcp message should be different from udp one. Ok, tomorrow I guess I'll fix it

**Comment by narkisr** [comment]
Ok, thanks

**Comment by s-mage** [comment]
@narkisr please look at the last commit, is it acceptable? I'm feeling like i'm rewriting all the library. Should I still make it as pull request or should I cancel it and make my own library?

**Comment by narkisr** [comment]
From reading the last commit it looks pretty nice, I don't think you need to separate this to another library 

How did you check that its working? did you use logstash and Kibana4?

**Comment by s-mage** [comment]
I use graylog itself. It still not finished and not tested at all :) I added [in progress] to header and will clean this up in the next couple of days if you don't mind.

**Comment by narkisr** [comment]
Sure go ahead, let me know when your done and ill test against kibana

Thanks

**Comment by s-mage** [comment]
@narkisr I tested tcp and it works pretty well, but can't test udp. I must be doing something wrong, but your initial version doesn't work for my server as well as my new one (I actually got your code for gzip/chunking and rewrote https://github.com/t0xa/gelfj/blob/master/src/main/java/org/graylog2/GelfUDPSender.java this one for sending and tried to combine them). So please test and/or fix it.

I'll fit timbre adapter.

**Comment by narkisr** [comment]
Ok, ill look into this and let you know if there are any changes required 

Thanks!

**Comment by s-mage** [comment]
Hi @narkisr, any luck with testing?

**Comment by narkisr** [comment]
@s-mage I'm on it, had a busy couple of weeks but ill attend to this on the weekend

**Comment by s-mage** [comment]
@narkisr that would be great!

**Comment by narkisr** [comment]
Ok, iv tested UDP (btw its quite easy you use [this](https://github.com/opskeleton/kibana4-sandbox) and run the midje facts) and it seems to work fine (I didn't test the TCP one yet). 

Iv also added a couple of comments on the pull request that id appreciate if you could apply

Thanks~!

File: src/gelfino/tcp.clj [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+(defn need-reconnect? [s ts]
+  (> (- ts (:last-op @(.last-operation s)))
+     (:timeout @(.last-operation s))))
+
+(defn update-last-operation [s]
+  (swap! (.last-operation s) assoc :last-op (timestamp)))
+
+(defn send-message [s msg-hash]
+  (let [socket (if (or (.isClosed @(.socket-atom s))
```
Comment (narkisr) [inline src/gelfino/tcp.clj:31]: Please extract this if into a separate function

File: src/gelfino/helpers.clj [hunk 2]
```diff
@@ -0,0 +1,19 @@
+(ns gelfino.helpers
```
Comment (narkisr) [inline src/gelfino/helpers.clj:1]: Please rename this ns into common (this is the common logic between udp and tcp implementations)

File: .lein-failures [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (narkisr) [inline .lein-failures:1]: I think that this file should be ignored

**Pull request merged**